starlingcaptureapi/file_util.py

```diff
@@ -56,3 +58,48 @@ def digest_sha256(self, file_path):
                 hasher.update(byte_block)
             return hasher.hexdigest()
         # TODO: handle error (image not found, etc.)
+
+    def cidv1(self, file_path):
+        """Generates the CIDv1 of a file, as determined by ipfs add.
+
+        Args:
+            file_path: the local path to a file
+
+        Returns:
+            CIDv1 in the canonical string format
+        """
+
+        if not os.path.exists(os.path.expanduser("~/.ipfs")):
+            proc = subprocess.run(
+                ["ipfs", "init"],
+                stdout=subprocess.PIPE,
+                stderr=subprocess.STDOUT,
+                text=True,
+            )
+            if proc.returncode != 0:
+                raise Exception(
+                    f"'ipfs init' failed with code {proc.returncode} and output:\n\n{proc.stdout}"
+                )
+
+            _logger.info("Created IPFS repo since it didn't exist")
+
+        proc = subprocess.run(
+            [
+                config.IPFS_BIN_PATH,
+                "add",
+                "--only-hash",
```

Comment:
@YurkoWasHere had a concern if we run 1GB files what the performance would be. Curious to know. If we're only hashing I imagine it'll still be fast.

Reply:
When I did some tests I got numbers like:
- 322M in ~2s
- 1.4G in 8s.
- 1 GiB had an avg. time of 5.7s (after warm-up runs):

```
Benchmark 1: ipfs add --only-hash --cid-version=1 -Q test.bin
  Time (mean ± σ):      5.745 s ±  0.169 s    [User: 8.427 s, System: 0.391 s]
  Range (min … max):    5.474 s …  5.991 s    10 runs
```

Reply:
Still concerned in general about this as we are comparing Local idle NvME to Cloud Storage.

But this is not a `IPFS` vs `IPFS-HASH-ONLY.js` question but a bigger question about server performance at scale.  
Same goes for all IO heavy actions (hashing, encrypting etc)